google/cloud/bigtable/table.py

```diff
@@ -132,6 +132,15 @@ def __init__(self, table_id, instance, mutation_timeout=None, app_profile_id=Non
         self._app_profile_id = app_profile_id
         self.mutation_timeout = mutation_timeout
 
+        self._table_impl = (
+            self._instance._client._data_client.get_table(
+                self._instance.instance_id,
+                self.table_id,
+            )
+            if self._instance
+            else None
```

Comment:
Why do you need `if self._instance`? It doesn't seem optional?

I don't think we'd ever want this to be None, right?